.githooks/pre-commit

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+readonly SWIFT_FILE_REGEX="\(.*\).swift$"
+
+function changed_swift_files() {
+    git diff \
+        --diff-filter=d \
+        --staged \
+        --name-only | grep -e "$SWIFT_FILE_REGEX"
+}
+
+changed_swift_files | while read swift_file;
+do
+    ./Scripts/format_code.sh "${swift_file}"
+
+    if [[ $? -ne 0 ]]
+    then
+        echo "Failed to process Swift file: ${swift_file}. Pre-commit hook failed"
+        exit 1
+    fi
+
+    git add "${swift_file}";
+done
```

Comment:
Actually, I think it would be good idea to use the fact that `swift-format` is capable of processing a list of input files at once and instead pass it an array of filenames to improve the performance of our script. I tested the following solution and it seems to be working just fine:
```
function changed_swift_files() {
    git diff \
        --diff-filter=d \
        --staged \
        --name-only \
        -- '*.swift' |
        awk '{ print "\""$0"\""; }' |
        tr '\n' ' '
}

SWIFT_FILES="$(changed_swift_files)"

if ! ./Scripts/format_code.sh "${SWIFT_FILES[@]}"
then
    echo "Pre-commit hook failed" 1>&2
    exit 1
fi

echo "${SWIFT_FILES[@]}" | xargs git add
```
And then in `format_code.sh` we would have this code:
```
SWIFT_FORMAT_INPUT_FILENAMES="./Package.swift ./Sources"

if [[ $# -ne 0 ]]
then
    SWIFT_FORMAT_INPUT_FILENAMES="$*"
fi

echo "${SWIFT_FORMAT_INPUT_FILENAMES[@]}" | xargs \
        swift-format format \
        --configuration ./.swift-format \
        --in-place \
        --recursive
```
It is an alternative approach, but I think it's worth considering.